include/engines/physics_info.h

```diff
@@ -35,87 +32,77 @@ class PhysicsInfo {
   int LocationMax = UINT16_MAX;  // default to this to have no ceiling ... 0 is
                                  // still hardcoded on the bottom
   float ThrustMax = 200;         // ???
-  int Mass = 1;                  // default to 1 so Thrust = Acceleration
-  int Gravity = 0;               // default to 0 for no gravity
-  float BounceFactor = 1.0;      // default to 1 for no bounce
-  int ExplodeVelocity = 0;       // default to 0 to explode on contact
+  int Mass = 1;                  // default to 1 so thrust = acceleration
+
+  float BounceFactor = 1.0;  // default to 1 for no bounce
+  int ExplodeVelocity = 0;   // default to 0 to explode on contact
 
   PhysicsInfo() { Reset(); }
 
   void RandomizeVelocityVector(float maxMagnitude) {
-    const float negMagnitude = maxMagnitude * -1;
-    const float hypotenuese = maxMagnitude * maxMagnitude;
-    do {
-      Velocity = random::Float(negMagnitude, maxMagnitude);
-      xVelocity = random::Float(negMagnitude, maxMagnitude);
-    } while (xVelocity * xVelocity + Velocity * Velocity >
-             hypotenuese);  // Repeat until we're within the unit circle
+    velocity = math::Vector2D<float>::RandomVector(maxMagnitude);
   }
 
   void Reset() {
-    Acceleration = 0;
-    Velocity = 0;
-    Location = 0;
-    Thrust = 0;
+    // Re-init physical vectors
+    velocity = math::Vector2D<float>();
+    location = math::Vector2D<float>();
+    thrust = math::Vector2D<float>();
+
+    // Reset collision detection
+    respect_edges = true;
     HasHitEdge = false;
     HasExploded = false;
-    Time = millis();
+
+    // Update time tracking
+    last_move_time = millis();
   }
 
-  void Move(bool respectEdges = true) {
-    if (Thrust > ThrustMax) {
-      Thrust = ThrustMax;
+  void Move() {
+    if (thrust.y > ThrustMax) {
+      thrust.y = ThrustMax;
     }
 
-    float timeDiff = (float)(millis() - Time) / 1000;
+    const auto now = millis();
+    const float timeDiff = (float)(now - last_move_time) / 1000.0f;

```

Comment:
what does the f do here?

Reply:
The f is a way that we can show the compiler that this literal value is meant to be interpreted as a float. I think the #.0 indicates that too but sometimes a compiler can optimize that away so I'm trying to be explicit that it's a float